controllers/datasciencecluster/datasciencecluster_controller.go

```diff
@@ -335,6 +335,10 @@ var configMapPredicates = predicate.Funcs{
 		if e.ObjectNew.GetName() == "prometheus" && e.ObjectNew.GetNamespace() == "redhat-ods-monitoring" {
 			return false
 		}
+		// Do not reconcile on kserver's inferenceservice-config CM updates, for rawdeployment
+		if e.ObjectNew.GetName() == "inferenceservice-config" && (e.ObjectNew.GetNamespace() == "redhat-ods-applications" || e.ObjectNew.GetNamespace() == "opendatahub") {
```

Comment:
Please correct if I'm wrong but I'm not sure if this if condition works like we need: during an upgrade (i.e. from version 2.1 to 2.2) this value should not be overridden otherwise the user will need to restore the configuration after every release.

@etirelli 
Is this the predicate that you mentioned be that can be used to bypass the reconcile?
Is there any way (like an annotation?) to let the user opt-out the reconcile? This configMap is something the user might want to update even if it is not something that we officially document yet

@vaibhavjainwiz 
FYI

Reply:
originally with this change, my thought goes with "changes in one specific CM should not lead to unnecessary reconcile of DSC"
but after some offline talk, I think the real problem we want to resolve here is:
user can update CM's data and not get reverted back during the same ODH release and even update ODH upgrade to a new release.

so i think in this case we will need 2 changes:
- prevent DSC reconcile because of this CM has been updated
- skip ownerreference on some resources
and the only thing i can think of for (2) is to use an annotation as indication, which user can add it when they are updating CM's data

Reply:
Out of curiosity, why the namespace test is needed?

Reply:
@israel-hdez 
> Out of curiosity, why the namespace test is needed?

the part to check if it is in opendatahub or in redhat-ods-applications?
only to narrow down if the CM is in these 2 namespaces. do you see it is possible that users create a different application namespace and deploy kseve into that?

Reply:
good point @israel-hdez 
if not set ownerreference, then users will need to cleanup these resource by themselves when they Removed component or uninstall ODH.
we need to be clear: 
- if annoation is set by the users, what consequence it will be
- if component manifests has it set by default, do we need to state this in the release docs to user 
leave above ^ to  @danielezonca
and i start to lean towards your idea to "keep ownerreference but just skip patch"
this is mainly because i am thinking, in the first case, annoation is post added by user, so when it first gets applied to the cluster (by component set as Managed) it should get ownerreference already. to skip it , only works the annotation has been added in the manifests out of the box.

Reply:
Well overall the use case is to let user customise this configMap and at that point the operator should not reconcile it anymore.
About the delete I agree that it is fine to keep removing it because it is tied to KServe so if the user remove KServe it is fine to remove it.

I'm fine with an exception (if) or an annotation to opt-out. The "skip patch" should work but we need to make sure it works also when we update from one version to another (i.e. 2.7 -> 2.8)

Note: this feature for now is a bit "advanced" and not really something we would like to document for "normal usage" but it can be necessary for hotfixes or ad hoc solutions and at the moment the only way to make it work (as far as I know) is to scale the operator to 0 that is definitely big as impact.

Reply:
i made another updates on this PR:
- we will keep adding owerreference (same logic as before)
- we do not patch resource if it has annotation set
- even upgrade to a new version of ODH, it consists with same customized content if annotation exists
- when component is Removed, such resource will be removed as well.

Reply:
This logic seems good to me (but I have not tested the PR)

Reply:
Why would you need to test it manually? Can we add some automated tests to cover those scenarios? Is it a big effort?